akka-stream-tests/src/test/scala/akka/stream/impl/fusing/InterpreterSpec.scala

```diff
@@ -569,39 +569,64 @@ class InterpreterSpec extends AkkaSpec with GraphInterpreterSpecKit {
 
   }
 
-  private[akka] case class Doubler[T]() extends PushPullStage[T, T] {
-    var oneMore: Boolean = false
-    var lastElem: T = _
-
-    override def onPush(elem: T, ctx: Context[T]): SyncDirective = {
-      lastElem = elem
-      oneMore = true
-      ctx.push(elem)
+  private[akka] case class Doubler[T]() extends GraphStage[FlowShape[T, T]] {
+    val out: Outlet[T] = Outlet("Doubler-out")
+    val in: Inlet[T] = Inlet("Doubler-in")
+
+    override def createLogic(inheritedAttributes: Attributes): GraphStageLogic = new GraphStageLogic(shape) {
+      var latest: T = _
+      var oneMore: Boolean = false
```

Comment:
Ping @drewhk . I assume `grab` is not allowed to return null, but there doesn't seem to be an assert nor documentation for it. Wdyt?